content/blog/2019/08/2019-08-17-folder-auth-plugin.adoc

```diff
@@ -0,0 +1,121 @@
+to be managed, permissions given to a folder through a folder role get inherited
+to all of its children. This is useful for giving access to multiple projects
+through a single role. Similarly, an agent role can be applied to multiple agents
+and assigned to multiple users.
+
+==== Performance Improvements over Role Strategy Plugin
+
+This plugin is designed to outperform Role Strategy Plugin in permission
+checks. The improvements were measured using the
+link:blog/2019/06/21/performance-testing-jenkins/[micro-benchmark framework]
+I had created during the first phase of my GSoC project.
+Benchmarks for identical configurations for both plugin show that the
+permissions check are up to 934x faster for 500 global roles when compared to
+the global roles from the Role Strategy 2.13, which in itself contains several
+performance improvements. Comparing folder roles with Role Strategy's project
+roles, a permission check for access to a job almost 15x faster for 250 projects
+organized in two-level deep folders on an instance with 150 users. You can see
+the benchmarks and the result comparisons
+link:https://github.com/jenkinsci/folder-auth-plugin/pull/13[here].
+
+==== Jenkins Configuration as Code Support
+
+The plugin supports Jenkins Configuration-as-Code so you can configure permissions
+without going through the Web UI. A YAML configuration looks like this:
+
+[source, yaml]
+----
+jenkins:
+  authorizationStrategy:
+    folderBased:
+      globalRoles:
+        - name: "admin"
+          permissions:
+            - id: "hudson.model.Hudson.Administer"
+              # ...
+          sids:
+            - "admin"
+        - name: "read"
+          permissions:
+            - id: "hudson.model.Hudson.Read"
+          sids:
+            - "user1"
+      folderRoles:
+        - folders:
+            - "root"
+          name: "viewRoot"
+          permissions:
+            - id: "hudson.model.Item.Read"
+          sids:
+            - "user1"
+      agentRoles:
+        - agents:
+            - "agent1"
+          name: "agentRole1"
+          permissions:
+            - id: "hudson.model.Computer.Configure"
+            - id: "hudson.model.Computer.Disconnect"
+          sids:
+            - "user1"
+----
+
+==== REST APIs with Swagger support
+
+The plugin provides REST APIs for managing roles with OpenAPI specifications
+through Swagger.json. You can check out the Swagger API on
+link:https://app.swaggerhub.com/apis/abhyudaya/folder-auth/1.0.0s[SwaggerHub].
+SwaggerHub provides stubs in multiple languages which can be downloaded and
+used to interact with the plugin. You can also see some sample requests from
+the command line using link:https://curl.haxx.se/[curl].
+
+image::/images/post-images/gsoc-folder-auth-plugin/swagger.png[alt=Screenshot of the APIs on SwaggerHub, height=500]
+
+image::/images/post-images/gsoc-folder-auth-plugin/swagger2.png[Another Screenshot of the APIs on SwaggerHub, height=400]
+
+== Links and Feedback
```

Comment:
Nice to have: What's next section? Maybe with references to UI improvements and so on

Reply:
Does this look good?

https://github.com/jenkins-infra/jenkins.io/pull/2410/files#diff-23c6bbdfd3aa65e648482a6a8064e12eR113-R118